bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -232,10 +313,160 @@ private Response getItemHistoryBean(String serviceName, String itemName, String
         for (Map.Entry<String, PersistenceService> service : persistenceServices.entrySet()) {
             ServiceBean serviceBean = new ServiceBean();
             serviceBean.name = service.getKey();
+            PersistenceService persistence = service.getValue();
+            if (persistence instanceof ModifiablePersistenceService) {
+                serviceBean.classname = "ModifiablePersistenceService";
+            } else if (persistence instanceof QueryablePersistenceService) {
+                serviceBean.classname = "QueryablePersistenceService";
+            } else {
+                serviceBean.classname = "PersistenceService";
+            }
 
             beanList.add(serviceBean);
         }
 
         return beanList;
     }
+
+    private Response getServiceItemList(String serviceName) {
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
+            // TODO: Add handler for default service once this is available in ESH
```

Comment:
I'm not sure if there is a reference. We had this discussion when I first added the persistence REST interface such that the user didn't need to request the service, and we would use the default. However there was no easy way to read the default service - I think this is only available in the persistence manager (from memory, which might be wrong).

We basically need a system wide method for reading the default persistence service.